.github/workflows/windows-meson-msys2.yml

```diff
@@ -2,27 +2,166 @@ name: MSYS2 builds
+            arch: x86_64
+            sys: MINGW64
+            max_warnings: 19
+          # - name: Clang x86
+          #   toolchain: clang-
+          #   arch: i686
+          #   sys: CLANG32
+          #   max_warnings: 19
+          - name: Clang x86_64
+            toolchain: clang-
+            arch: x86_64
+            sys: CLANG64
+            max_warnings: 19
+          - name: GCC +tests
+            toolchain: ""
             arch: x86_64
             sys: MINGW64
+            run_tests: true
+            max_warnings: -1
+          - name: GCC +debugger
+            toolchain: ""
+            arch: x86_64
+            sys: MINGW64
+            max_warnings: 21
+            build_flags: -Denable_debugger=normal
     
     defaults:
       run:
         shell: msys2 {0}
     
     steps:
+      - name: Disable git line ending conversion
+        shell: powershell
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
+
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      
+      - name: Setup MSYS2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{matrix.conf.sys}}
+          update: true
+          install: >-
+            git
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-meson
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-gcc
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-ccache
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-pkgconf
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-python
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-ntldd
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-ncurses
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-glib2
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-fluidsynth
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-munt-mt32emu
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-libpng
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-opusfile
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-SDL2
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-SDL2_net
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-zlib
+      
+      - name:  Prepare compiler cache
+        id:    prep-ccache
+        run: |
+          mkdir -p "${CCACHE_DIR}"
+          echo "::set-output name=dir::C:\ccache"
+          echo "::set-output name=today::$(date -I)"
+          echo "::set-output name=yesterday::$(date --date=yesterday -I)"
+          echo "::set-output name=name_hash::$(echo '${{ matrix.conf.name }}' | shasum | cut -b-8)"
+
+      - uses:  actions/cache@v2
+        id:    cache-ccache
+        with:
+          path: ${{ steps.prep-ccache.outputs.dir }}
```

Comment:
The top-level `CCACHE_DIR = ...` environment variable lets us set it in one place, and use it every where that's needed.  It would be nice if we didn't have to hardcode the Windows-equivalent for it as `c:\ccache`, because then this means we have two representations of the same path.

Do you know if `cygpath` is any help?

`echo "::set-output name=dir::$(cygpath -w $CCACHE_DIR)"`

This assumes that the GitHub shell is given the MSYS2 path and can run the `cygpath` tool.. it's work a try anyway, I think?

Reply:
I'll give it a try. That `run` is operating inside the MSYS2 shell, but I don't know how that interacts with `::set-output`